legend-engine-core/legend-engine-core-executionPlan-execution/legend-engine-executionPlan-dependencies/src/main/java/org/finos/legend/engine/plan/dependencies/domain/date/PureDate.java

```diff
@@ -704,7 +704,7 @@ public PureDate addMonths(int months)
         return copy;
     }
 
-    public PureDate addWeeks(int weeks)
+    public PureDate addWeeks(long weeks)
```

Comment:
We should use Math.multiplyExact(7L, weeks) here to avoid the possibility of overflow.